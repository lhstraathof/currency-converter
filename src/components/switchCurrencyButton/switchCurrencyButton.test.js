import { fireEvent, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { SwitchCurrencyButton } from '@components';

const testData = {
	action: jest.fn(),
};

describe('SwitchCurrencyButton', () => {
	it('renders correctly', () => {
		render(<SwitchCurrencyButton handleButtonClick={testData.action} />);
		expect(screen.getByTestId('switchCurrencyButton')).toBeInTheDocument();
	});

	it('fires clickAction onClick', () => {
		const { getByTestId } = render(<SwitchCurrencyButton handleButtonClick={testData.action} />);
		const button = getByTestId('switchCurrencyButton');

		fireEvent.click(button);
		expect(testData.action).toHaveBeenCalledTimes(1);
	});

	it('matches snapshot', () => {
		const component = create(<SwitchCurrencyButton handleButtonClick={testData.action} />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
