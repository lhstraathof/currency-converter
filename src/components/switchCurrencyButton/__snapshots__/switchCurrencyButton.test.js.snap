// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchCurrencyButton matches snapshot 1`] = `
<button
  className="switchCurrencyButton"
  data-testid="switchCurrencyButton"
  onClick={[MockFunction]}
>
  <svg>
    exchange-icon.svg
  </svg>
</button>
`;
